policyengine_uk/variables/gov/hmrc/income_tax/allowances/savings_allowance/savings_allowance.py

```diff
@@ -0,0 +1,47 @@
+from policyengine_uk.model_api import *
+
+class savings_allowance(Variable):
+    value_type = float
+    entity = Person
+    label = "Savings Allowance for the year"
+    definition_period = YEAR
+    reference = "Income Tax Act 2007 s. 12B"
+    unit = GBP
+
+    def formula(person, period, parameters):
+        amounts = parameters(
+            period
+        ).gov.hmrc.income_tax.allowances.personal_savings_allowance
+
+        savings_allowance_category = person("savings_allowance_category", period)
+        max_allowance = amounts[savings_allowance_category.decode_to_str().lower()]
```

Comment:
if we want to incorporate this type of structure, we need a breakdown parameter - we should consolidate all three parameters into one where each possible `savings_allowance_category` is defined with teh respective value

Reply:
we then dont need the `.decode_to_str`